src/redis-cli.c

```diff
@@ -2973,7 +2973,7 @@ static void clusterManagerOptimizeAntiAffinity(clusterManagerNodeArray *ipnodes,
         /* We'll try to randomly swap a slave's assigned master causing
          * an affinity problem with another random slave, to see if we
          * can improve the affinity. */
-        int rand_idx = rand() % offending_len;
+        int rand_idx = offending_len > 1 ? rand() % offending_len : 0;
```

Comment:
If offending_len is 0, doesn't the next line access the offenders array out of range?

Reply:
Zero index in `offenders` always exists in this case because it allocated according to `cluster_manager.nodes->len` which definitely greater than 0

Reply:
i'm sorry, i'm not familiar with that code, but something doesn't feel right to me. maybe you can help me figure it out.
it looks like the code doesn't expect `offending_len` to be 0, and solving the division by zero might just lead to another error down the line.
like you said, access violation into that array isn't possible, but still the `first` and `second` pointers might not point to the right thing.
I have a that maybe `if (score == 0) break;` should be extended to catch this case, or alternatively `clusterManagerGetAntiAffinityScore` should be modified to return `0` when `offending_len` is set to `0`.
can you look into this for me?

Reply:
OK, I'll try to figure it out today :)

Reply:
@oranagra you was right, when `offending_len` is 0 `offenders` doesn't contain valid list of replicas.
Also I found that score calculation in `clusterManagerGetAntiAffinityScore` works wrong, if I understood idea correctly.

Reply:
@yatsukhnenko i'm not familiar with this code.
for an easier review, can you try to explain under what circumstance was it buggy, what are the implications of the bug and what you fixed?
thanks.

Reply:
@oranagra I also don't really familiar with this code :smile:
I played with redis cluster in kubernetes and wanted to create cluster using only one node but `redis-cli` requires minimum three addresses so I decided to use the same address three times. That's how I found division by zero bug.
After your review I looked at `clusterManagerGetAntiAffinityScore` and was trying to understand what's happening there.
According to description score for three master nodes on the same host have to be 30000 but it was 60000 so I found that reason is in using `sdscatprintf` with the same sds argument twice.